agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/detail/DetailState.kt

```diff
@@ -0,0 +1,20 @@
+package com.nibalk.tasky.agenda.presentation.detail
+
+import com.nibalk.tasky.agenda.presentation.model.AgendaType
+import java.time.LocalDate
+import java.time.LocalTime
+
+data class DetailState(
+    // From Args
+    val isEditingMode: Boolean = false,
+    val selectedDate: LocalDate = LocalDate.now(),
+    val agendaId: String = "",
+    val agendaType: AgendaType = AgendaType.TASK,
+    // From backend
+    val title: String = "",
+    val description: String = "",
+    val startDate: LocalDate = LocalDate.now(),
+    val startTime: LocalTime = LocalTime.now(),
+    val endDate: LocalDate = LocalDate.now(),
+    val endTime: LocalTime = LocalTime.now(),
```

Comment:
Could there maybe be a better structure for this state, so the state only contains those endDate/endTime fields if you're actually dealing with an event? That way, you'd communicate by design what the specifics of each agenda item are.